### PR TITLE
04/01/2021, model association

### PR DESCRIPTION
**Pull request by haupt-2855**
WHAT (optional)
- Create model association
Screenshot from 2022-01-06 15-24-20

### CODE REVIEW

**Review by hieupp-1207 — APPROVED** [review]

**Comment by haupt-2855** [comment]
ready

File: app/models/category.rb [hunk 1]
```diff
@@ -1,2 +1,7 @@
 class Category < ApplicationRecord
+  has_many :product, dependent: :destroy
+  has_many :child, class_name: :Category,
```
Comment (vanvtt-0952) [inline app/models/category.rb]:
```suggestion
  has_many :children, class_name: Category.name,
```

File: app/models/category.rb [hunk 2]
```diff
@@ -1,2 +1,7 @@
...
+    foreign_key: :parent_id, dependent: :destroy
+  belongs_to :parent, class_name: :Category,
+    foreign_key: :parent_id, optional: true
```
Comment (vanvtt-0952) [inline app/models/category.rb]:
```suggestion
  belongs_to :parent, class_name: Category.name, optional: true
```

File: app/models/order.rb [hunk 3]
```diff
@@ -1,2 +1,4 @@
 class Order < ApplicationRecord
+  belongs_to :user
+  has_many :order_detail, dependent: :destroy
```
Comment (vanvtt-0952) [inline app/models/order.rb]:
```suggestion
  has_many :order_details, dependent: :destroy
```

File: app/models/category.rb [hunk 4]
```diff
@@ -1,2 +1,7 @@
 class Category < ApplicationRecord
+  has_many :product, dependent: :destroy
```
Comment (vanvtt-0952) [inline app/models/category.rb]:
```suggestion
  has_many :products, dependent: :destroy
```

File: app/models/product.rb [hunk 5]
```diff
@@ -1,2 +1,5 @@
 class Product < ApplicationRecord
+  has_many :order_detail, dependent: :destroy
+  has_many :rating, dependent: :destroy
```
Comment (vanvtt-0952) [inline app/models/product.rb]:
```suggestion
  has_many :order_details, dependent: :destroy
  has_many :ratings, dependent: :destroy
```

File: app/models/user.rb [hunk 6]
```diff
@@ -1,2 +1,4 @@
 class User < ApplicationRecord
+  has_many :order, dependent: :destroy
+  has_many :rating, dependent: :destroy
```
Comment (vanvtt-0952) [inline app/models/user.rb]:
```suggestion
  has_many :order, dependent: :destroy
  has_many :ratings, dependent: :destroy
```

File: db/migrate/20220105063048_create_orders.rb [hunk 7]
```diff
@@ -1,6 +1,7 @@
 class CreateOrders < ActiveRecord::Migration[6.1]
   def change
     create_table :orders do |t|
+      t.integer :user_id
```
Comment (vanvtt-0952) [inline db/migrate/20220105063048_create_orders.rb]: gen migration mới chứ bạn

**Review by hantt-2856 — APPROVED** [review]

**Comment by haupt-2855** [comment]
ready

**Pull request merged**